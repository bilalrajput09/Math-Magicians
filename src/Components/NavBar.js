import { NavLink } from 'react-router-dom';
import styles from './NavBar.module.css';

const NavBar = () => {
  const myLinks = [
    {
      id: 1,
      text: 'Home',
      path: '/',
    },
    {
      id: 2,
      text: 'Calculator',
      path: '/calculator',
    },
    {
      id: 3,
      text: 'Quote',
      path: '/quote',
    },
  ];

  return (
    <header className={styles.header}>
      <h1>Math-Magicians</h1>
      <nav>
        <ul className={styles.links_contianer}>
          {myLinks.map((link) => (
            <li key={link.id}>
              <NavLink to={link.path}>{link.text}</NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default NavBar;
