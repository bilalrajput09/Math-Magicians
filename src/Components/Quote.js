import { useEffect, useState } from 'react';
import styles from './Quote.module.css';

const Quote = () => {
  const [qouteResult, setQouteResult] = useState([]);
  const [loadingState, setLoadingState] = useState(true);
  const [errorState, setErrorState] = useState(false);

  const category = 'computers';

  useEffect(() => {
    const fetchData = async () => {
      try {
        const res = await fetch(
          `https://api.api-ninjas.com/v1/quotes?category=${category}`,
          {
            method: 'GET',
            withCredentials: true,
            headers: {
              'X-Api-Key': 'mTAX8mjo76+AlFPlXASyCg==WSfxrthcCwbMDsPX',
              'Content-Type': 'application/json',
            },
          },
        );
        setLoadingState(false);
        setErrorState(false);
        const json = await res.json();
        setQouteResult(json[0]);
      } catch (err) {
        setErrorState(true);
        setLoadingState(false);
        setQouteResult([]);
      }
    };

    fetchData();
  }, []);

  return (
    <div data-testid="main-container">
      <div className={styles.quote_container}>
        <h1>{loadingState && 'LOADING...'}</h1>
        <h3>{!errorState && !loadingState && `"${qouteResult.quote}"`}</h3>
        <span className={styles.author}>
          {!loadingState && qouteResult.author}
        </span>
        <span className={styles.error}>
          {errorState && 'Something went wrong!'}
        </span>
      </div>
    </div>
  );
};

export default Quote;
