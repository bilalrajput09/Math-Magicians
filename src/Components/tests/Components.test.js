import { screen, render, waitFor } from '@testing-library/react';
import { Route, MemoryRouter } from 'react-router-dom';
import Quote from '../Quote';
import NavBar from '../NavBar';

describe('Tests for Components', () => {
  test('Check the element on the page', async () => {
    render(<Quote />);
    await waitFor(() => {
      expect(screen.getByTestId('main-container')).toMatchSnapshot();
    });
  });

  test('Check the navlink', () => {
    <Route>
      (render(
      <NavBar />
      )
      const linkElement = screen.getByText(/Maths Magician/i);
      expect(linkElement).toMatchSnapshot();)
    </Route>;
  });

  test('Check click event', () => {
    <MemoryRouter>
      render(
      <NavBar />
      )
      const linkElement = screen.getByText(/Home/i); const
      expectedResult = getByText(/Welcome to our page!/i);
      fireEvent.click(linkElement); expect(expectedResult).toBeInTheDocument();
    </MemoryRouter>;
  });
});
