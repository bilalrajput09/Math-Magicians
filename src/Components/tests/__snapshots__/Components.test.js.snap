// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Components Check the element on the page 1`] = `
<div
  data-testid="main-container"
>
  <div
    class="quote_container"
  >
    <h1>
      LOADING...
    </h1>
    <h3 />
    <span
      class="author"
    />
    <span
      class="error"
    />
  </div>
</div>
`;

exports[`Tests for Components Verify the content on Calculator page! 1`] = `
<h1>
  Yeah do some math!
</h1>
`;

exports[`Tests for Components Verify the content on home page! 1`] = `
<h2>
  Welcome to our page!
</h2>
`;
